source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    void displayMarkerPose(const Zivid::PointCloud &pointCloud,
+                           const std::vector<cv::Point2f> &corners2D,
+                           const std::vector<cv::Point2f> &center2D)
+    {
+        const auto bgr = pointCloudToCvBGR(pointCloud);
+
+        auto poseCenter = center2D[0];
+        auto xAxisPoint = cv::Point2f(((corners2D[3].x + corners2D[0].x) / 2), ((corners2D[3].y + corners2D[0].y) / 2));
+        auto yAxisPoint = cv::Point2f(((corners2D[2].x + corners2D[3].x) / 2), ((corners2D[2].y + corners2D[3].y) / 2));
+
+        cv::circle(bgr, poseCenter, 2, cv::Scalar(255, 0, 0), 2);
+        cv::arrowedLine(bgr, poseCenter, xAxisPoint, cv::Scalar(0, 0, 255), 2);
+        cv::arrowedLine(bgr, poseCenter, yAxisPoint, cv::Scalar(0, 255, 0), 2);
+
+        cv::imshow("Pose of aruco marker", bgr);
+        cv::waitKey(0);
+    }
+
+    Zivid::Matrix4x4 getMarkerPose(const Zivid::PointCloud &pointCloud, const aruconano::Marker &markers)
+    {
+        // Extracting 2d corners and estimateing 2d center
+        const auto corners2D = get2DMarkerCorners(markers);
+        const auto center2D = estimate2DMarkerCenters(markers);
+
+        // estimating 3d corners and center from 2d data
+        const auto corners3D = estimate3DMarkerPoints(pointCloud, corners2D);
+        const auto center3D = estimate3DMarkerPoints(pointCloud, center2D);
+
+        // extracting origin and calcualting normal vectors for x-, y- and z-axis
+        auto origin = cv::Vec3d{ center3D[0].x, center3D[0].y, center3D[0].z };
+
+        auto xAxis = cv::Vec3d{ corners3D[2].x - corners3D[1].x,
+                                corners3D[2].y - corners3D[1].y,
+                                corners3D[2].z - corners3D[1].z };
+
+        auto yAxis = cv::Vec3d{ corners3D[0].x - corners3D[1].x,
+                                corners3D[0].y - corners3D[1].y,
+                                corners3D[0].z - corners3D[1].z };
+
+        auto u = xAxis / cv::norm(xAxis, cv::NORM_L2);
+        auto v = yAxis / cv::norm(yAxis, cv::NORM_L2);
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotationMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotationMatrix.at<double>(i, 0) = u[i];
+            rotationMatrix.at<double>(i, 1) = v[i];
+            rotationMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        auto transform = transformationMatrix(rotationMatrix, origin);
```

Comment:
this can go directly to return - line 227

Reply:
yes it can. Moved it